explorer/interpreter/interpreter.cpp

```diff
@@ -689,7 +689,7 @@ auto Interpreter::Convert(Nonnull<const Value*> value,
           break;
         case Value::Kind::StaticArrayType: {
           const auto& array_type = cast<StaticArrayType>(*destination_type);
-          destination_element_types.resize(array_type.size(),
+          destination_element_types.resize(tuple->elements().size(),
```

Comment:
I would expect this to be asserting that the array size matches the tuple size; it shouldn't matter which you use. Does that occur now?

Reply:
The reason I uses tuple's size is that array_type's size seems not initialized at this moment, which is weird for me because in later context array's size is able to be accessed.

Reply:
This change has been reverted since the previous commits didn't acutally do type deduction, and its size is not inferred, too.